src/extensions/default/CodeFolding/main.js

```diff
@@ -371,6 +407,11 @@ define(function (require, exports, module) {
         Editor.forEveryEditor(function (editor) {
             CodeMirror.commands.unfoldAll(editor._codeMirror);
             removeGutter(editor);
+            //disconnect any mutation observers on the gutter
+            if (gutterObservers[editor.document.file.fullPath]) {
```

Comment:
`gutterObservers[editor.document.file.fullPath]` could be cached to an variable too